### PR TITLE
fix pruner bugs and add model compression README

### PR DESCRIPTION
**Pull request by tanglang96**
fix pruner bugs and add model compression README

### CODE REVIEW

File: examples/model_compress/main_torch_pruner.py [hunk 1]
```diff
@@ -43,52 +44,53 @@ def test(model, device, test_loader):
...
-    optimizer = torch.optim.SGD(model.parameters(), lr = 0.01, momentum = 0.5)
+    optimizer = torch.optim.SGD(model.parameters(), lr=0.01, momentum=0.5)
     for epoch in range(10):
         print('# Epoch {} #'.format(epoch))
         train(model, device, train_loader, optimizer)
         test(model, device, test_loader)
-        
-        pruner.update_epoch(epoch)
+
+        pruner.update_epoch(epoch + 1)
```
Comment (liuzhe-lz) [inline examples/model_compress/main_torch_pruner.py]: Let's discuss whether epoch should start from 0 or 1 next time.

File: src/sdk/pynni/nni/compression/torch/builtin_pruners.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import torch
 from .compressor import Pruner
 
-__all__ = ['LevelPruner', 'AGP_Pruner', 'SensitivityPruner']
+__all__ = ['LevelPruner', 'AGP_Pruner']
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/builtin_pruners.py]: you removed sensitivitypruner, so please update doc accordingly

File: src/sdk/pynni/nni/compression/torch/builtin_pruners.py [hunk 3]
```diff
@@ -17,14 +17,22 @@ def __init__(self, config_list):
...
+    def calc_mask(self, weight, config, op_name, **kwargs):
+        if self.if_init_list.get(op_name, True):
+            w_abs = weight.abs()
+            k = int(weight.numel() * config['sparsity'])
+            if k == 0:
+                return torch.ones(weight.shape).type_as(weight)
+            threshold = torch.topk(w_abs.view(-1), k, largest=False).values.max()
+            mask = torch.gt(w_abs, threshold).type_as(weight)
+            self.mask_list.update({op_name: mask})
+            self.if_init_list.update({op_name: False})
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/builtin_pruners.py]: suggest to change False to True

File: src/sdk/pynni/nni/compression/torch/builtin_pruners.py [hunk 4]
```diff
@@ -17,14 +17,22 @@ def __init__(self, config_list):
...
 
-    def calc_mask(self, weight, config, **kwargs):
-        w_abs = weight.abs()
-        k = int(weight.numel() * config['sparsity'])
-        if k == 0:
-            return torch.ones(weight.shape).type_as(weight)
-        threshold = torch.topk(w_abs.view(-1), k, largest=False).values.max()
-        return torch.gt(w_abs, threshold).type_as(weight)
+    def calc_mask(self, weight, config, op_name, **kwargs):
+        if self.if_init_list.get(op_name, True):
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/builtin_pruners.py]: if op_name not in self.if_init_list

File: examples/model_compress/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# Run model compression examples
+
+You can run these examples easily like this, take torch pruning for example
+
+```bash
+python main_torch_pruner.py
+```
+
+Model compression can be configured in 2 ways
```
Comment (QuanluZhang) [inline examples/model_compress/README.md]: This example uses AGP Pruner. Initiating a pruner needs a user provided configuration which can be provided in two ways:

File: examples/model_compress/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+- By reading ```configure_example.yaml```, this can make codes clean when your configuration is complicated
```
Comment (QuanluZhang) [inline examples/model_compress/README.md]: codes -> code

File: examples/model_compress/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    'final_sparsity': 0.8,
+    'start_epoch': 1,
+    'end_epoch': 11,
+    'frequency': 1,
+    'op_type': 'default'
+}]
+pruner = AGP_Pruner(configure_list)
+```
+
+Please notice that when ```pruner(model)``` called, our model compression codes will be **automatically injected** and you can fine-tune your model **without** any modifications, masked weights **won't** be updated any more during fine tuning.
```
Comment (QuanluZhang) [inline examples/model_compress/README.md]: When `pruner(model)` is called, your model is injected with masks as embedded operations. For example, a layer takes a weight as input, we will insert an operation between the weight and the layer, this operation takes the weight as input and outputs a new weight applied by the mask. Thus, the masks are applied at any time the computation goes through the operations. You can fine-tune your model **without** any modifications.

File: examples/model_compress/README.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+```python
+for epoch in range(10):
+    print('# Epoch {} #'.format(epoch))
+    train(model, device, train_loader, optimizer)
+    test(model, device, test_loader)
+    pruner.update_epoch(epoch + 1)
+```
+
```
Comment (QuanluZhang) [inline examples/model_compress/README.md]: `update_epoch` is for pruner to be aware of epochs, so that it could adjust masks during training.

File: docs/en_US/Compressor/Overview.md [hunk 9]
```diff
@@ -8,7 +8,6 @@ We have provided two naive compression algorithms and four popular ones for user
 |---|---|
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Overview.md]: “We have provided two naive compression algorithms and four popular ones for users, including three pruning algorithms and three quantization algorithms:” this line should also be updated

File: docs/en_US/Compressor/Pruner.md [hunk 10]
```diff
@@ -48,7 +48,7 @@ from nni.compression.tensorflow import AGP_Pruner
 config_list = [{
     'initial_sparsity': 0,
     'final_sparsity': 0.8,
-    'start_epoch': 1,
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Pruner.md]: what is the meaning of start_epoch=0, end_epoch=10?

Comment (tanglang96) [reply docs/en_US/Compressor/Pruner.md]: @QuanluZhang start_epoch=0, end_epoch=10 means pruning starts from epoch 0 and ends at 10, previous default start_epoch=1 in the algorithm but we usually start from 0, so I modified them all.
